src/Fakta/Status.fs

```diff
@@ -1,9 +1,44 @@
 ﻿module Fakta.Status
+open Fakta
+open Fakta.Logging
+open Fakta.Impl
+open System
+open System.Collections
+open NodaTime
+open HttpFs.Client
+open Chiron
+
+
+let faktaStatusString = "Fakta.status"
+
+let statusDottedPath (funcName: string) =
+  (sprintf "%s.%s" faktaStatusString funcName)
+
 
 /// Leader is used to query for a known leader 
-let leader (s : FaktaState) : Async<Choice<string, Error>> =
-  raise (TBD "TODO")
+let leader (state : FaktaState) : Async<Choice<string, Error>> =  
+    let urlPath = "leader"
+    let uriBuilder = UriBuilder.ofStatus state.config urlPath
+    let result = Async.RunSynchronously (call state (statusDottedPath urlPath) id uriBuilder HttpMethod.Get)
+    async {
```

Comment:
The reason all of these functions are returning async is to avoid the above blocking calls. Doing it like doesn't add anything.